### DISCUSSION TITLE
midi Thru

### DISCUSSION THREAD

**Discussion by jelhoste** [tttapa/Control-Surface#778] [Q&A] [answered]
Hello, thank you very much for this library!
I want to activate the midiThru for an interface, so that what enters on Rx is retransmitted on Tx of this same interface (like "turnThruOn()" in the Arduino Midi library of FortySevenEffects).
I have not been able to find the procedure from the documentation, could you tell me where to find this information or tell me how to do it please?
Thanks,

Jean Eudes L'Hoste

**Comment by tttapa** [comment, marked as answer]
You can use a MIDI_Pipe and pipe the interface into itself. See e.g. [MIDI Tutorial: Routing MIDI messages](https://tttapa.github.io/Control-Surface-doc/Doxygen/d3/df7/midi-tutorial.html#midi_md-routing)

```cpp
midi >> mpipe >> midi;
```

**Comment by jelhoste** [comment]
Ok, thanks !
have a good day !
